extrabees/src/main/java/binnie/extrabees/worldgen/WorldGenHiveMarble.java

```diff
@@ -31,19 +31,18 @@ public boolean generate(World world, Random rand, BlockPos position) {
 			return false;
 		}
 
-		//generate when one face is different from marble
+		//generate when at least two faces are different from marble
 		int otherFace = 0;
 		for (EnumFacing face : EnumFacing.values()) {
 			if (!world.getBlockState(position.offset(face)).getBlock().equals(blockAtPos)) {
 				otherFace++;
 				if (otherFace > 1) {
+					world.setBlockState(position, ExtraBees.hive.getDefaultState().withProperty(BlockExtraBeeHives.HIVE_TYPE, EnumHiveType.MARBLE));
 					return true;
 				}
 			}
 		}
 
-		world.setBlockState(position, ExtraBees.hive.getDefaultState().withProperty(BlockExtraBeeHives.HIVE_TYPE, EnumHiveType.MARBLE));
-
-		return true;
+		return false;
```

Comment:
I assume true/false indicates if anything was changed/generated so I used true only after actually generating a hive and false otherwise.